src/Microsoft.SqlTools.ServiceLayer/Utility/CommandOptions.cs

```diff
@@ -26,18 +28,31 @@ public CommandOptions(string[] args)
                     string arg = args[i];
                     if (arg.StartsWith("--") || arg.StartsWith("-"))
                     {
+                        // Extracting arguments and properties
                         arg = arg.Substring(1).ToLowerInvariant();
-                        switch (arg)
+                        string argName = arg;
+                        string argProperty = "";
+                        int splitIndex = arg.IndexOf(' ');
```

Comment:
Ah, I see what's going on now. I think it might be a bit easier to have your argument set up like `--locale=es` which would give you a single argument for the locale. Then in the locale case you could split on = and take the second item in the array